pkg/utils/k8sutils.go

```diff
@@ -171,3 +180,23 @@ func IsForbiddenError(err error) bool {
 	}
 	return false
 }
+
+// This function takes raw YAML string and converts it to Kubernetes Tolerations array
+func YamlToTolerations(raw string) ([]v1.Toleration, error) {
+	if raw == "" {
+		return []v1.Toleration{}, nil
+	}
+
+	rawJSON, err := yaml.ToJSON([]byte(raw))
+	if err != nil {
+		return []v1.Toleration{}, err
+	}
+
+	var tolerations []v1.Toleration
+	err = json.Unmarshal(rawJSON, &tolerations)
+	if err != nil {
+		return []v1.Toleration{}, err
+	}
+
+	return tolerations, nil
+}
```

Comment:
As this is a temporary solution, let's make all of the changes inlined in the controller file